src/rgw/rgw_rest_s3.cc

```diff
@@ -3429,7 +3429,10 @@ int RGWCopyObj_ObjStore_S3::get_params(optional_yield y)
     s->info.args.get_bool(RGW_SYS_PARAM_PREFIX "copy-if-newer", &copy_if_newer, false);
   }
 
-  copy_source = s->info.env->get("HTTP_X_AMZ_COPY_SOURCE");
+  const char *copy_source_temp = s->info.env->get("HTTP_X_AMZ_COPY_SOURCE");
```

Comment:
you can probably make `copy_source` an optional string, and in `RGWPutObj_ObjStore_S3::send_response()` check on it instead of checking `copy_source.empty()`

Reply:
@yuvalif  but that means there are subsequent changes we need to make as in [rgw_op.h](https://github.com/ceph/ceph/blob/main/src/rgw/rgw_op.h#LL1196C1-L1196C27) as `copy_source` is defined as `std::string` not `boost::optional<const std::string&>`?

Reply:
ok. lets keep the change here small